Framework/DataHandling/src/LoadSpiceXML2DDet.cpp

```diff
@@ -460,6 +476,80 @@ LoadSpiceXML2DDet::parseSpiceXML(const std::string &xmlfilename) {
   return vecspicenode;
 }
 
+//----------------------------------------------------------------------------------------------
+/// parse binary integer file
+std::vector<unsigned int>
+LoadSpiceXML2DDet::binaryParseIntegers(std::string &binary_file_name) {
+  // check binary file size
+  ifstream infile(binary_file_name.c_str(), ios::binary);
+  streampos begin, end;
+  begin = infile.tellg();
+  infile.seekg(0, ios::end);
+  end = infile.tellg();
+  g_log.information() << "File size is: " << (end - begin) << " bytes.\n";
+
+  size_t num_unsigned_int =
+      static_cast<size_t>(end - begin) / sizeof(unsigned int);
+  size_t num_dets = num_unsigned_int - 2;
+  g_log.information() << "File contains " << num_unsigned_int
+                      << " unsigned integers and thus " << num_dets
+                      << " detectors.\n";
+
+  // define output vector
+  std::vector<unsigned int> vec_counts(num_dets);
+  infile.seekg(0, ios::beg);
+
+  // read each integer... time consuming
+  //  int max_count = 0;
+  // char buffer[sizeof(int)];
+  unsigned int buffer;
+  unsigned int total_counts(0);
+
+  // read detector size (row and column)
+  infile.read((char *)&buffer, sizeof(buffer));
+  size_t num_rows = static_cast<size_t>(buffer);
+  infile.read((char *)&buffer, sizeof(buffer));
+  size_t num_cols = static_cast<size_t>(buffer);
+
```

Comment:
Since you have the `num_dets` from the file size and the `num_rows` and `num_cols` read from the file this is a good point to add a check that `num_rows * num_cols == num_dets` before you try and read the rest of the file and throw with a useful message if it doesn't.

Reply:
good point!